### ISSUE TITLE
Celery v5.5.0 Release

### ISSUE THREAD

**Issue by Nusnus** [celery/celery#9140]
# Minor Release Overview: v5.5.0

This issue will summarize the status and discussion in preparation for the new release. It will be used to track the progress of the release and to ensure that all the necessary steps are taken. It will serve as a checklist for the release and will be used to communicate the status of the release to the community.

> ⚠️ **Warning:** The release checklist is a living document. It will be updated as the release progresses. Please check back often to ensure that you are up to date with the latest information.

## Checklist
- [x] Codebase Stability
- [x] Breaking Changes Validation
- [x] Compile Changelog
- [ ] Release
- [ ] Release Announcement

# Release Details
The release manager is responsible for completing the release end-to-end ensuring that all the necessary steps are taken and that the release is completed in a timely manner. This is usually the owner of the release issue but may be assigned to a different maintainer if necessary.

- Release Manager: Tomer Nosrati (@Nusnus)
- Release Date: TBD
- Release Branch: `main`

# Release Steps
The release manager is expected to execute the checklist below. The release manager is also responsible for ensuring that the checklist is updated as the release progresses. Any changes or issues should be communicated under this issue for centralized tracking.

# Potential Release Blockers

## 1. Codebase Stability
- [ ] The `main` branch build passes

    [![Build Status](https://github.com/celery/celery/actions/workflows/python-package.yml/badge.svg)](https://github.com/celery/celery/actions/workflows/python-package.yml)

## 2. Breaking Changes Validation
A patch release should not contain any breaking changes. The release manager is responsible for reviewing all of the merged PRs since the last release to ensure that there are no breaking changes. If there are any breaking changes, the release manager should discuss with the maintainers to determine the best course of action if an obvious solution is not apparent.

## 3. Compile Changelog
The release changelog is set in two different places:
1. The [Changelog.rst](https://github.com/celery/celery/blob/main/Changelog.rst) that uses the RST format.
2. The GitHub Release auto-generated changelog that uses the Markdown format. This is auto-generated by the GitHub Draft Release UI.

> ⚠️ **Warning:** The pre-commit changes should not be included in the changelog.

To generate the changelog automatically, [draft a new release](https://github.com/celery/celery/releases/new) on GitHub using a fake new version tag for the automatic changelog generation. Notice the actual tag creation is done **on publish** so we can use that to generate the changelog and then delete the draft release without publishing it thus avoiding creating a new tag.

- Create a new tag
<img width="502" alt="CleanShot 2023-09-05 at 22 06 24@2x" src="https://github.com/celery/celery/assets/4662342/69d7eb78-0dd1-4a37-9266-d718e2340fad">

- Generate Markdown release notes
<img width="389" alt="CleanShot 2023-09-05 at 22 13 39@2x" src="https://github.com/celery/celery/assets/4662342/83c6a4b6-1676-496e-8d75-5a6b10352b5b">

- Copy the generated release notes.

- Delete the draft release without publishing it.

### 3.1 Changelog.rst
Once you have the actual changes, you need to convert it to rst format and add it to the [Changelog.rst](https://github.com/celery/celery/blob/main/Changelog.rst) file. The new version block needs to follow the following format:
```rst
.. _version-x.y.z:

x.y.z
=====

:release-date: YYYY-MM-DD HH:MM P.M/A.M TimeZone
:release-by: Release Manager Name

Changes list in RST format.
```

These changes will reflect in the [Change history](https://docs.celeryq.dev/en/stable/changelog.html) section of the documentation.

### 3.2 Changelog PR
The changes to the [Changelog.rst](https://github.com/celery/celery/blob/main/Changelog.rst) file should be submitted as a PR. This will PR should be the last merged PR before the release.

### 4.1 Prepare releasing environment
Before moving forward with the release, the release manager should ensure that bumpversion and twine are installed. These are required to publish the release.

### 4.2 Bump version
The release manager should bump the version using the following command:
```bash
bumpversion patch
```
The changes should be pushed directly to main by the release manager.

At this point, the git log should appear somewhat similar to this:
```
commit XXX (HEAD -> main, tag: vX.Y.Z, upstream/main, origin/main)
Author: Release Manager
Date:   YYY

    Bump version: a.b.c → x.y.z

commit XXX
Author: Release Manager
Date:   YYY

    Added changelog for vX.Y.Z (#1234)
```
If everything looks good, the bump version commit can be directly pushed to `main`:
```bash
git push origin main --tags
```

### 4.3 Publish release to PyPI
The release manager should publish the release to PyPI using the following commands running under the root directory of the repository:
```bash
python setup.py clean build sdist bdist_wheel
```
If the build is successful, the release manager should publish the release to PyPI using the following command:
```bash
twine upload dist/celery-X.Y.Z*
```

> ⚠️ **Warning:** The release manager should double check that the release details are correct (project/version) before publishing the release to PyPI.

> ⚠️ **Critical Reminder:** Should the released package prove to be faulty or need retraction for any reason, do not delete it from PyPI. The appropriate course of action is to "yank" the release.

## Release Announcement
After the release is published, the release manager should create a new GitHub Release and set it as the latest release.

<img width="401" alt="CleanShot 2023-09-05 at 22 51 24@2x" src="https://github.com/celery/celery/assets/4662342/6edcba33-b6a9-4242-85e8-8260a2ede868">

### Add Release Notes
On a per-case basis, the release manager may also attach an additional release note to the auto-generated release notes. This is usually done when there are important changes that are not reflected in the auto-generated release notes.

### OpenCollective Update
After successfully publishing the new release, the release manager is responsible for announcing it on the project's OpenCollective [page](https://opencollective.com/celery/updates). This is to engage with the community and keep backers and sponsors in the loop.

**Comment by Nusnus** [comment]
[v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.

**Comment by JockeTF** [comment]
I've been playing around with quorum queues, and they seem really nice! I had set up a small RabbitMQ 3.13 cluster using a simplified (upgraded, without the entrypoint script, and with manual cluster joining) version of [serkodev/rabbitmq-cluster-docker](serkodev/rabbitmq-cluster-docker). I threw 65 536 tasks at it over the cause of an hour while this little thing was running.

```bash
while true; do
  for i in {1..3}; do
    docker kill "rabbitmq$i"
    sleep 30
    docker start "rabbitmq$i"
    sleep 30
  done
done
```

It seems 16 tasks were lost from two of the kills (I don't know why yet), but the system remained stable. The worst thing I saw beyond that was a handful of our response times slowing down by half a second after a kill. I also noticed that non-default queues for tasks were being created as classic queues instead of quorum queues.

Thanks for this release! It is looking pretty great so far!

**Comment by Nusnus** [comment]
> [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.

[Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.

**Comment by connorlwilkes** [comment]
What is the current ETA for this release? We are hitting the stuck worker issue with Airflow and celery and would be interested in upgrading to fix that

**Comment by Nusnus** [comment]
> > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> 
> [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.

[Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.

**Comment by Nusnus** [comment]
Celery v5.5.0b4 will be released this week.
RC -> Final will be done during October.
The due date for Python 3.8/3.13 is the 1st of October, and we will try to support both in v5.5.

**Comment by Nusnus** [comment]
> > > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> > 
> > 
> > [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.
> 
> [Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.

[Celery v5.5.0b4](https://github.com/celery/celery/releases/tag/v5.5.0b4) released.

**Comment by Nusnus** [comment]
> Celery v5.5.0b4 will be released this week. RC -> Final will be done during October. The due date for Python 3.8/3.13 is the 1st of October, and we will try to support both in v5.5.

Unless we receive a significant enough contribution, these are the estimated deadlines for the subsequent release cycles:
- Release Candidate 1: Between 6th and 12th of October (next week).
- Release Candidate 2: ~1 week afterward (Mid-October).
- Final Release: ~1 week afterward.

So v5.5.0 should be released between the 20th and 26th of October.

That being said, if there will be only very minor changes after RC1 and it won’t make sense to go with RC2, we will consider RC1 to be stable enough and release the last few changes (e.g., English typo fixes in the docs) with the final release, and skip RC2 altogether. The deadline for the final release will remain the same though, and RC1 will have more “air time” instead.

**Comment by Nusnus** [comment]
> What is the current ETA for this release? We are hitting the stuck worker issue with Airflow and celery and would be interested in upgrading to fix that

@connorlwilkes FYI my last new updates above

**Comment by Nusnus** [comment]
> > > > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> > > 
> > > 
> > > [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.
> > 
> > 
> > [Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.
> 
> [Celery v5.5.0b4](https://github.com/celery/celery/releases/tag/v5.5.0b4) released.

[Celery v5.5.0rc1](https://github.com/celery/celery/releases/tag/v5.5.0rc1) released.

**Comment by Nusnus** [comment]
> > Celery v5.5.0b4 will be released this week. RC -> Final will be done during October. The due date for Python 3.8/3.13 is the 1st of October, and we will try to support both in v5.5.
> 
> Unless we receive a significant enough contribution, these are the estimated deadlines for the subsequent release cycles:
> - Release Candidate 1: Between 6th and 12th of October (next week).
> - Release Candidate 2: ~1 week afterward (Mid-October).
> - Final Release: ~1 week afterward.
> 
> So v5.5.0 should be released between the 20th and 26th of October.
> 
> That being said, if there will be only very minor changes after RC1 and it won’t make sense to go with RC2, we will consider RC1 to be stable enough and release the last few changes (e.g., English typo fixes in the docs) with the final release, and skip RC2 altogether. The deadline for the final release will remain the same though, and RC1 will have more “air time” instead.

RC2 will be delayed to next week.
Considering adding RC3, will update next week.
This will push the final release to the very end of October.

The motivation is to allow contributors enough space to finalize the current open issues so we can release the next pre-release without any known issues.

**Comment by Nusnus** [comment]
> The motivation is to allow contributors enough space to finalize the current open issues so we can release the next pre-release without any known issues.

Few of these are:
- https://github.com/celery/celery/pull/9302
- https://github.com/celery/celery/pull/9207
- https://github.com/celery/celery/pull/9348
- https://github.com/celery/kombu/issues/2157

And hopefully but optionally,
- https://github.com/celery/kombu/pull/2052

**Comment by Nusnus** [comment]
> > The motivation is to allow contributors enough space to finalize the current open issues so we can release the next pre-release without any known issues.
> 
> Few of these are:
> 
> * [Add native incr support for GCSBackend #9302](https://github.com/celery/celery/pull/9302)
> * [Native Delayed Delivery in RabbitMQ #9207](https://github.com/celery/celery/pull/9207)
> * [fix(perform_pending_operations): prevent task duplication on shutdown… #9348](https://github.com/celery/celery/pull/9348)
> * [Regression in 5.5.0rc1: sqlalchemy create_engine issue kombu#2157](https://github.com/celery/kombu/issues/2157)
> 
> And hopefully but optionally,
> 
> * [Add support for Python 3.13 kombu#2052](https://github.com/celery/kombu/pull/2052)

@thedrow @auvipy please prioritize your attention to the above issues so I can continue with the release procedure.
Also notice my [previous comment](https://github.com/celery/celery/issues/9140#issuecomment-2419519621).

Thanks!

**Comment by vachillo** [comment]
let me know if this should be logged elsewhere.

we are still seeing the same redis connection issues with the latest `rc1` release. The celery worker fails to reconnect to redis after raising `SERVER_CLOSED_CONNECTION_ERROR` and it doesn't seem to reconnect.

python 3.11 
celery 5.5.0rc1
kombu 5.4.2

trace:
```
Traceback (most recent call last):
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/worker/consumer/consumer.py\", line 340, in start
    blueprint.start(self)
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/bootsteps.py\", line 116, in start
    step.start(parent)
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/worker/consumer/consumer.py\", line 759, in start
    c.loop(*c.loop_args())
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/worker/loops.py\", line 97, in asynloop
    next(loop)
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/hub.py\", line 308, in create_loop
    poll_timeout = fire_timers(propagate=propagate) if scheduled else 1
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/hub.py\", line 149, in fire_timers
    entry()
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/timer.py\", line 70, in __call__
    return self.fun(*self.args, **self.kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/timer.py\", line 137, in _reschedules
    return fun(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/transport/redis.py\", line 554, in maybe_restore_messages
    return channel.qos.restore_visible(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/transport/redis.py\", line 410, in restore_visible
    with Mutex(client, self.unacked_mutex_key,
  File \"/usr/local/lib/python3.11/contextlib.py\", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/transport/redis.py\", line 166, in Mutex
    lock_acquired = lock.acquire(blocking=False)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/lock.py\", line 210, in acquire
    if self.do_acquire(token):
       ^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/lock.py\", line 226, in do_acquire
    if self.redis.set(self.name, token, nx=True, px=timeout):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/commands/core.py\", line 2333, in set
    return self.execute_command(\"SET\", *pieces, **options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/ddtrace/contrib/redis/patch.py\", line 153, in _instrumented_execute_command
    return _run_redis_command(ctx=ctx, func=func, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/ddtrace/contrib/redis/patch.py\", line 130, in _run_redis_command
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 548, in execute_command
    return conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 65, in call_with_retry
    fail(error)
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 552, in <lambda>
    lambda error: self._disconnect_raise(conn, error),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 538, in _disconnect_raise
    raise error
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 62, in call_with_retry
    return do()
           ^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 549, in <lambda>
    lambda: self._send_command_parse_response(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 524, in _send_command_parse_response
    conn.send_command(*args)
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 476, in send_command
    self.send_packed_command(
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 449, in send_packed_command
    self.check_health()
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 441, in check_health
    self.retry.call_with_retry(self._send_ping, self._ping_failed)
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 67, in call_with_retry
    raise error
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 62, in call_with_retry
    return do()
           ^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 431, in _send_ping
    if str_if_bytes(self.read_response()) != \"PONG\":
                    ^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 512, in read_response
    response = self._parser.read_response(disable_decoding=disable_decoding)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/resp2.py\", line 15, in read_response
    result = self._read_response(disable_decoding=disable_decoding)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/resp2.py\", line 25, in _read_response
    raw = self._buffer.readline()
          ^^^^^^^^^^^^^^^^^^^^^^^
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/socket.py\", line 115, in readline
    self._read_from_socket()
  File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/socket.py\", line 68, in _read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
```

**Comment by thedrow** [comment]
> let me know if this should be logged elsewhere. there is a [discussion](https://github.com/celery/celery/discussions/9359) with the same issue/tech stack described.
> 
> we are still seeing the same redis connection issues with the latest `rc1` release. The celery worker fails to reconnect to redis after raising `SERVER_CLOSED_CONNECTION_ERROR` and it doesn't seem to reconnect.
> 
> python 3.11 celery 5.5.0rc1 kombu 5.4.2
> 
> trace:
> 
> ```
> Traceback (most recent call last):
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/worker/consumer/consumer.py\", line 340, in start
>     blueprint.start(self)
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/bootsteps.py\", line 116, in start
>     step.start(parent)
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/worker/consumer/consumer.py\", line 759, in start
>     c.loop(*c.loop_args())
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/celery/worker/loops.py\", line 97, in asynloop
>     next(loop)
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/hub.py\", line 308, in create_loop
>     poll_timeout = fire_timers(propagate=propagate) if scheduled else 1
>                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/hub.py\", line 149, in fire_timers
>     entry()
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/timer.py\", line 70, in __call__
>     return self.fun(*self.args, **self.kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/asynchronous/timer.py\", line 137, in _reschedules
>     return fun(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/transport/redis.py\", line 554, in maybe_restore_messages
>     return channel.qos.restore_visible(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/transport/redis.py\", line 410, in restore_visible
>     with Mutex(client, self.unacked_mutex_key,
>   File \"/usr/local/lib/python3.11/contextlib.py\", line 137, in __enter__
>     return next(self.gen)
>            ^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/kombu/transport/redis.py\", line 166, in Mutex
>     lock_acquired = lock.acquire(blocking=False)
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/lock.py\", line 210, in acquire
>     if self.do_acquire(token):
>        ^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/lock.py\", line 226, in do_acquire
>     if self.redis.set(self.name, token, nx=True, px=timeout):
>        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/commands/core.py\", line 2333, in set
>     return self.execute_command(\"SET\", *pieces, **options)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/ddtrace/contrib/redis/patch.py\", line 153, in _instrumented_execute_command
>     return _run_redis_command(ctx=ctx, func=func, args=args, kwargs=kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/ddtrace/contrib/redis/patch.py\", line 130, in _run_redis_command
>     result = func(*args, **kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 548, in execute_command
>     return conn.retry.call_with_retry(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 65, in call_with_retry
>     fail(error)
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 552, in <lambda>
>     lambda error: self._disconnect_raise(conn, error),
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 538, in _disconnect_raise
>     raise error
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 62, in call_with_retry
>     return do()
>            ^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 549, in <lambda>
>     lambda: self._send_command_parse_response(
>             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/client.py\", line 524, in _send_command_parse_response
>     conn.send_command(*args)
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 476, in send_command
>     self.send_packed_command(
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 449, in send_packed_command
>     self.check_health()
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 441, in check_health
>     self.retry.call_with_retry(self._send_ping, self._ping_failed)
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 67, in call_with_retry
>     raise error
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/retry.py\", line 62, in call_with_retry
>     return do()
>            ^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 431, in _send_ping
>     if str_if_bytes(self.read_response()) != \"PONG\":
>                     ^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/connection.py\", line 512, in read_response
>     response = self._parser.read_response(disable_decoding=disable_decoding)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/resp2.py\", line 15, in read_response
>     result = self._read_response(disable_decoding=disable_decoding)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/resp2.py\", line 25, in _read_response
>     raw = self._buffer.readline()
>           ^^^^^^^^^^^^^^^^^^^^^^^
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/socket.py\", line 115, in readline
>     self._read_from_socket()
>   File \"/opt/webapp/api/.venv/lib/python3.11/site-packages/redis/_parsers/socket.py\", line 68, in _read_from_socket
>     raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
> redis.exceptions.ConnectionError: Connection closed by server.
> ```

Most people confirmed it is fixed. This must be another bug.
Can you file an issue about this?

**Comment by Nusnus** [comment]
> > The motivation is to allow contributors enough space to finalize the current open issues so we can release the next pre-release without any known issues.
> 
> Few of these are:
> 
> * [Add native incr support for GCSBackend #9302](https://github.com/celery/celery/pull/9302)
> * [Native Delayed Delivery in RabbitMQ #9207](https://github.com/celery/celery/pull/9207)
> * [fix(perform_pending_operations): prevent task duplication on shutdown… #9348](https://github.com/celery/celery/pull/9348)
> * [Regression in 5.5.0rc1: sqlalchemy create_engine issue kombu#2157](https://github.com/celery/kombu/issues/2157)
> 
> And hopefully but optionally,
> 
> * [Add support for Python 3.13 kombu#2052](https://github.com/celery/kombu/pull/2052)

Unfortunately, the remaining issues (above and more) were not closed before the expected deadline, so we’ll postpone the release date accordingly.

Fortunately, this _was_ completed on time:
* https://github.com/celery/kombu/pull/2052

Remaining Steps/Issues:
1. **Kombu:** https://github.com/celery/kombu/pull/2167
2. **Kombu:** Release v5.5.0rc2
3. **Celery:** https://github.com/celery/celery/pull/9207
4. **Celery:** Release v5.5.0rc2
5. Fix critical issues only.
6. **Kombu:** Release v5.5.0
7. **Celery:** Release v5.5.0

### Estimated New Release Date
2nd week of November.

**Comment by Nusnus** [comment]
> Kombu: Release v5.5.0rc2

@thedrow please check current `main` of Kombu againsts the latest `main` of Celery and run all of the unit, integration and smoke tests. Once you confirm here that our both `main` branches play along, I’ll release Kombu v5.5.0rc2 and complete my own release checklist to make sure the version is viable for the next RC pre-release.

Once this is done, you (@thedrow) can update #9207 and we can focus on getting it merged so I can complete the rest of the release steps.

@auvipy FYI

Thanks! 🙌

**Comment by thedrow** [comment]
@Nusnus All tests are passing. You can release 5.5.0rc2.

**Comment by Nusnus** [comment]
> @Nusnus All tests are passing. You can release 5.5.0rc2.

Thank you!
Done.
- v5.5.0rc2 is released: https://github.com/celery/kombu/releases/tag/v5.5.0rc2
- Celery Bump: https://github.com/celery/celery/pull/9382

I also found a new bug in Celery `main`/rc1 during my release checklist, compared to v5.4.0.
See #9383 for more info.

**Comment by Nusnus** [comment]
### Remaining Steps/Issues (29 Oct 2024)
1. **Celery:** https://github.com/celery/celery/pull/9383
2. **Celery:** https://github.com/celery/celery/pull/9207
3. **Celery:** Release v5.5.0rc2
4. Fix critical issues only.
5. **Kombu:** Release v5.5.0
6. **Celery:** Release v5.5.0

### Estimated New Release Date
2nd week of November.

**Comment by Nusnus** [comment]
### Remaining Steps/Issues (29 Oct 2024)
1. **Celery:** https://github.com/celery/celery/pull/9383
2. **Celery:** https://github.com/celery/celery/issues/9385
3. **Celery:** https://github.com/celery/celery/pull/9207
4. **Celery:** Release v5.5.0rc2
5. Fix critical issues only.
6. **Kombu:** Release v5.5.0
7. **Celery:** Release v5.5.0

### Estimated New Release Date
2nd week of November.

**Comment by Nusnus** [comment]
> ### Remaining Release Blockers (29 Oct 2024)
> 1. **Celery:** [Missing CPendingDeprecationWarning from Celery v5.4.0 #9383](https://github.com/celery/celery/issues/9383)
> 2. **Celery:** [Bug: Log "Global QoS is disabled. Prefetch count in now static.” appears for non-amqp transports #9385](https://github.com/celery/celery/issues/9385)
> 3. **Celery:** [Native Delayed Delivery in RabbitMQ #9207](https://github.com/celery/celery/pull/9207)
> 4. **Celery:** Release v5.5.0rc2
> 5. Fix critical issues only.
> 6. **Kombu:** Release v5.5.0
> 7. **Celery:** Release v5.5.0
> 
> ### Estimated New Release Date
> ~2nd~ 4th week of November.
> 
> @thedrow @auvipy FYI

All of the release blockers are gone!! Woohoo! 🥳

## New release schedule
- **Mon 2024-11-18:** Celery v5.5.0rc2 (using Kombu [v5.5.0rc2](https://github.com/celery/kombu/releases/tag/v5.5.0rc2))
- **Mon 2024-12-02:** Kombu v5.5.0.
- **Mon 2024-12-02:** Celery v5.5.0.

@thedrow @auvipy I want to declare a complete code freeze from tomorrow’s RC2 release except for bug/doc fixes.
The two weeks between RC2 and the release should be strictly reviewed (for both Kombu and Celery). Anything noncritical should be postponed to the next version using the [v5.5.1 milestone](https://github.com/celery/celery/milestone/43). Dependency upgrades can only be accepted for patch releases, etc.

If there’s any exception, I’d appreciate it if you could tag me for review so I can be on top of the entire release process.
This is one hell of a version :)

**Comment by Nusnus** [comment]
> > > > > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> > > > 
> > > > 
> > > > [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.
> > > 
> > > 
> > > [Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.
> > 
> > 
> > [Celery v5.5.0b4](https://github.com/celery/celery/releases/tag/v5.5.0b4) released.
> 
> [Celery v5.5.0rc1](https://github.com/celery/celery/releases/tag/v5.5.0rc1) released.

[v5.5.0rc2](https://github.com/celery/celery/releases/tag/v5.5.0rc2) released.

**Comment by Nusnus** [comment]
Due to a new release blocker (#9433), a few more minor issues that need to be resolved and being unavailable for a few days to fix the blocker, I am postponing the release by another week, from 2nd to 9th of Dec.

@thedrow @auvipy FYI

**Comment by Nusnus** [comment]
> Due to a new release blocker (#9433), a few more minor issues that need to be resolved and being unavailable for a few days to fix the blocker, I am postponing the release by another week, from 2nd to 9th of Dec.
> 
> @thedrow @auvipy FYI

More issues are being opened about bugs in the current RC.
This is good, but it needs to be fixed before the release.

Due to the great feedback from the community and the increased scope of issues to handle, I am proposing a new release deadline to take advantage of the high-quality acceptance tests this new version is receiving (on Kombu’s end as well!).

## New release schedule
- **Mon 2024-12-12:** Celery v5.5.0 Release Candidate 4.
- **Mon 2024-12-23:** Kombu v5.5.0.
- **Mon 2024-12-23:** Celery v5.5.0.

**Comment by Nusnus** [comment]
I also want to apologize for the very long release process.
This is our most significant release since 2021 IIRC and fortunately, the community is very active in reporting issues.
It is important to me that these issues be addressed to respect the effort that was made to report them. That is why I am adjusting the deadline, although I know many users are waiting for this release.

**Comment by auvipy** [comment]
it's alright. stability over speed this time!

**Comment by hfpf** [comment]
> > Due to a new release blocker (#9433), a few more minor issues that need to be resolved and being unavailable for a few days to fix the blocker, I am postponing the release by another week, from 2nd to 9th of Dec.
> > @thedrow @auvipy FYI
> 
> More issues are being opened about bugs in the current RC. This is good, but it needs to be fixed before the release.
> 
> Due to the great feedback from the community and the increased scope of issues to handle, I am proposing a new release deadline to take advantage of the high-quality acceptance tests this new version is receiving (on Kombu’s end as well!).
> 
> ## New release schedule
> * **Thu 2024-12-12:** Celery v5.5.0 Release Candidate 4.
> * **Mon 2024-12-23:** Kombu v5.5.0.
> * **Mon 2024-12-23:** Celery v5.5.0.

Is there any temporary workaround for this issue? https://github.com/celery/celery/discussions/7276#discussion-3851077

**Comment by Nusnus** [comment]
> Thu 2024-12-12: Celery v5.5.0 Release Candidate 4.

Our CI broke in the last 24 hours in `main`. I assume it is due to a patch release of a dep that dropped support with Python 3.8 where we probably use without pinning a version.
I’ll fix this and release the next RC ASAP next week.

**Comment by davidnaumann-bastian** [comment]
```python
@app.task(pydantic=True)
def example(
    pydantic_models: List[MyPydanticModel]
):
```

For importing a list of Pydantic models seeing the items in the `pydantic_models` list being dictionaries still (on rc3). Is this supposed to be how this functions (if so I can serialize it by hand but figured I'd bring it up in-case it is a bug)?

**Comment by Nusnus** [comment]
> > Thu 2024-12-12: Celery v5.5.0 Release Candidate 4.
> 
> Our CI broke in the last 24 hours in `main`. I assume it is due to a patch release of a dep that dropped support with Python 3.8 where we probably use without pinning a version. I’ll fix this and release the next RC ASAP next week.

> > > > > > > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> > > > > > 
> > > > > > 
> > > > > > [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.
> > > > > 
> > > > > 
> > > > > [Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.
> > > > 
> > > > 
> > > > [Celery v5.5.0b4](https://github.com/celery/celery/releases/tag/v5.5.0b4) released.
> > > 
> > > 
> > > [Celery v5.5.0rc1](https://github.com/celery/celery/releases/tag/v5.5.0rc1) released.
> > 
> > 
> > [Celery v5.5.0rc2](https://github.com/celery/celery/releases/tag/v5.5.0rc2) released.
> 
> [Celery v5.5.0rc3](https://github.com/celery/celery/releases/tag/v5.5.0rc3) released.

[Celery v5.5.0rc4](https://github.com/celery/celery/releases/tag/v5.5.0rc4) released.

That being done, let me expand upon the current status of things.

## Open Issues
> I also want to apologize for the very long release process. This is our most significant release since 2021 IIRC and fortunately, the community is very active in reporting issues. It is important to me that these issues be addressed to respect the effort that was made to report them. That is why I am adjusting the deadline, although I know many users are waiting for this release.

I’ve been tackling most blockers and quick/high-importance issues. The remaining issues are under my investigation to evaluate the urgency/importance and offer remediation.

Thank you for the great feedback - it seems the list of issues is finally shrinking :)

## Missing Release Docs
> This is our most significant release since 2021 IIRC

Unfortunately, the documentation of the new release does not cover many of the introduced changes, which are critical to mention when upgrading to v5.5. These changes include many new features, enhancements, bug fixes, **and caveats to be aware of**, including significant upgrades to internal mechanisms and improvements to many of our integrations.

I’ve already gone through almost all of the changelog and documented the most major changes, but it is not “production ready” yet, and I need to complete the missing updates.

## Other Priorities
In addition to the above, I am also working on the "Celery 2025 Roadmap”, that will shift the entire organization forward with Celery v6. Celery is a huge ship to steer, and this task is consuming most of my available time.
The relevancy of this roadmap is extremely high to this release, as my own focus will also steer accordingly, which means that if I don’t attend to the remaining issues in this release phase, they might be added to the endless years+ issues without any activity.

## Deadlines
> * ~**Mon 2024-12-23:** Kombu v5.5.0.~
> * ~**Mon 2024-12-23:** Celery v5.5.0.~

The release will be postponed to next year so I can also rest and recharge during the upcoming holidays. Thank you for following my updates and for your great support. I will use this time to reflect on the organization and return with fresh energy.

The next update will be in 2025, so I’ll wish everyone a happy new year already 🥳

**Comment by Nusnus** [comment]
> ```python
> @app.task(pydantic=True)
> def example(
>     pydantic_models: List[MyPydanticModel]
> ):
> ```
> 
> For importing a list of Pydantic models seeing the items in the `pydantic_models` list being dictionaries still (on rc3). Is this supposed to be how this functions (if so I can serialize it by hand but figured I'd bring it up in-case it is a bug)?

@mathiasertl

**Comment by mathiasertl** [comment]
> > ```python
> > @app.task(pydantic=True)
> > def example(
> >     pydantic_models: List[MyPydanticModel]
> > ):
> > ```
> > 
> > For importing a list of Pydantic models seeing the items in the `pydantic_models` list being dictionaries still (on rc3). Is this supposed to be how this functions (if so I can serialize it by hand but figured I'd bring it up in-case it is a bug)?
> 
> @mathiasertl 

Hi @Nusnus ,

Yes. Having models in generics makes the whole implementation a lot more complex. Consider that then you'd also have to consider many other types (`dict[str, MyPydanticModel]`) or even nested types (`list[dict[str, dict[str, MyPydanticModel]]]`)... Right?

In any of these cases, you can use a pydantic model directly 
 In case of a list, use a custom pydantic Root Model.

I'm open to reviewing an implementation for that however, of course!

Kr, Mati

**Comment by Javimenendez** [comment]
> > > ```python
> > > @app.task(pydantic=True)
> > > def example(
> > >     pydantic_models: List[MyPydanticModel]
> > > ):
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > For importing a list of Pydantic models seeing the items in the `pydantic_models` list being dictionaries still (on rc3). Is this supposed to be how this functions (if so I can serialize it by hand but figured I'd bring it up in-case it is a bug)?
> > 
> > 
> > @mathiasertl
> 
> Hi @Nusnus ,
> 
> Yes. Having models in generics makes the whole implementation a lot more complex. Consider that then you'd also have to consider many other types (`dict[str, MyPydanticModel]`) or even nested types (`list[dict[str, dict[str, MyPydanticModel]]]`)... Right?
> 
> In any of these cases, you can use a pydantic model directly In case of a list, use a custom pydantic Root Model.
> 
> I'm open to reviewing an implementation for that however, of course!
> 
> Kr, Mati

Hi @mathiasertl ,

I think this can be solved using TypeAdapter function from pydantic.

Here is a small example:

```python
from pydantic import BaseModel, TypeAdapter
from typing import Optional


class MyPydanticModel(BaseModel):
    foo: Optional[str] = None
    bar: Optional[str] = None
    

data={"foo":"foo_value", "bar":"bar_value"}


type_adapter= TypeAdapter(list[MyPydanticModel])
arg_value= type_adapter.validate_python([data])
# [MyPydanticModel(foo='foo_value', bar='bar_value')]

type_adapter= TypeAdapter(list[dict[str, dict[str, MyPydanticModel]]])
arg_value= type_adapter.validate_python([{"data":{"my_pydantic_model":data}}])
# [{'data': {'my_pydantic_model': MyPydanticModel(foo='foo_value', bar='bar_value')}}]
```

**Comment by mathiasertl** [comment]
> I think this can be solved using TypeAdapter function from pydantic.

I guess that would work as well. Thanks for the interesting approach!

My recommendation would be a custom root model. From pydantic instead:

https://docs.pydantic.dev/latest/concepts/models/#rootmodel-and-custom-root-types

Kr, Mat

**Comment by Nusnus** [comment]
> The release will be postponed to next year so I can also rest and recharge during the upcoming holidays. Thank you for following my updates and for your great support. I will use this time to reflect on the organization and return with fresh energy.
> 
> The next update will be in 2025, so I’ll wish everyone a happy new year already 🥳

Hello,

Unfortunately, a [security incident](https://github.com/celery/celery/discussions/9525) has halted our entire organization. Our team has been focused on resolving this issue, which has inevitably delayed the release cycle further. I recognize the wait for this release has been exceptionally long.

Thank you for your patience.

Normal activities should resume shortly.

**Comment by omerkatzkcg** [comment]
@Nusnus What are the current blockers for this release?

**Comment by violuke** [comment]
How is the recovery from the incident going? We're really looking forward to this release 🙏  (largely for the RabbitMQ Quorum Queues and Pydantic features 👀 )

I notice the [milestone](https://github.com/celery/celery/milestone/42) has more open issues on it than it did before (at least, I think it does). Anything that the community can do to help?

**Comment by Nusnus** [comment]
@violuke
> I notice the [milestone](https://github.com/celery/celery/milestone/42) has more open issues on it than it did before (at least, I think it does)

Unfortunately, it is not in sync with the actual plan, so the remaining gap for the release is not defined accurately for the 5.5 milestone.

> Anything that the community can do to help?

Yes, definitely. Most important are bugs in features **confirmed** to be relevant to production environments:

- Stamping Feature: #9446 and #9451
- Canvas Error Handling: #9107 (while keeping an eye on #6792)

Another issue was introduced between different RC releases:
- #9425: Requires validation on the issue + proposed solution.

About #6792: Not an actual blocker, but might be an easy win during the fix of #9107. 

Regarding the rest of the issues, I am considering re-evaluating and postponing to the next patch release.

### Fixing the stamping and the canvas bugs will trigger the next RC!

> How is the recovery from the incident going? We're really looking forward to this release 🙏 (largely for the RabbitMQ Quorum Queues and Pydantic features 👀 )

I am glad to share we are moving forward in full force. I was able to proceed with my 2025 Roadmap (mentioned [here](https://github.com/celery/celery/issues/9140#issuecomment-2555484070)) and start laying the ground for the upcoming asyncio support (shh!! 🤫). Communication with the community is also being reformed to allow better transparency. I am also working on solving our most difficult maintenance friction points to bring (at least) x10 performance boost in community support, response time, and even more! Multiple announcements are cooking up.. 

P.S
I am prioritizing the roadmap over the release items to create conditions that never allow such delays to happen in the first place, but I will treat any contribution for the above issues as super-critical and give my full attention. Tag me away :)

**Comment by Nusnus** [comment]
Stamping blocker fixed:
- https://github.com/celery/celery/pull/9575
- https://github.com/celery/celery/pull/9578

**Comment by Nusnus** [comment]
Canvas blocker fixed:
- https://github.com/celery/celery/pull/9580

Next RC will be released this week.

**Comment by Nusnus** [comment]
> Another issue was introduced between different RC releases:
> 
> * [Possible bug in 5.5.0rc2 -- app.start(["worker"]) error #9425](https://github.com/celery/celery/discussions/9425): Requires validation on the issue + proposed solution.
> 

Can’t reproduce with v5.5.0rc4/`main` per my attempt [here](https://github.com/celery/celery/discussions/9425#discussioncomment-12312345).
Not closing as “completed”, but removing from release blockers.

**Comment by Nusnus** [comment]
> > > > > > > > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> > > > > > > 
> > > > > > > 
> > > > > > > [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.
> > > > > > 
> > > > > > 
> > > > > > [Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.
> > > > > 
> > > > > 
> > > > > [Celery v5.5.0b4](https://github.com/celery/celery/releases/tag/v5.5.0b4) released.
> > > > 
> > > > 
> > > > [Celery v5.5.0rc1](https://github.com/celery/celery/releases/tag/v5.5.0rc1) released.
> > > 
> > > 
> > > [Celery v5.5.0rc2](https://github.com/celery/celery/releases/tag/v5.5.0rc2) released.
> > 
> > 
> > [Celery v5.5.0rc3](https://github.com/celery/celery/releases/tag/v5.5.0rc3) released.
> 
> [Celery v5.5.0rc4](https://github.com/celery/celery/releases/tag/v5.5.0rc4) released.

[Celery v5.5.0rc5](https://github.com/celery/celery/releases/tag/v5.5.0rc5) released.

**Comment by bityob** [comment]
Hey @Nusnus, 

Any idea when Celery 5.5 might be released?

Trying to plan ahead. 

Thanks! 🙌

**Comment by Nusnus** [comment]
> Hey @Nusnus, 
> 
> Any idea when Celery 5.5 might be released?
> 
> Trying to plan ahead. 
> 
> Thanks! 🙌

Last week of March (this month).

@auvipy fyi

**Comment by tallyoh** [comment]
Hello, Is the 5.5 release still on track for this week? Really looking forward to quorum queues.

**Comment by Nusnus** [comment]
@tallyoh 
> Hello, Is the 5.5 release still on track for this week? Really looking forward to quorum queues.

I'll have to stretch my statement to sun/mon (still March 😉), but yes. There are no new delays since my last update.

Finishing with this release will finally free up more capacity for the asyncio effort and more interesting stuff, so I can't wait to get it done already personally 🙂

**Comment by Nusnus** [comment]
Loading…
- https://github.com/celery/celery/pull/9644

**Comment by auvipy** [comment]
> Loading…
> - https://github.com/celery/celery/pull/9644

Nusnus, there is a hotfix in kombu main branch. You can include that in kombu 5.5.2 for stable celery 5.5. Best of luck

**Comment by Nusnus** [comment]
> > Loading…
> > - https://github.com/celery/celery/pull/9644
> 
> Nusnus, there is a hotfix in kombu main branch. You can include that in kombu 5.5.2 for stable celery 5.5. Best of luck

Yeah I saw your comment somewhere else and did it already in https://github.com/celery/celery/pull/9643 🙂

**Comment by Nusnus** [comment]
@athariri 
> [@Nusnus](https://github.com/Nusnus)
> 
> We've been running 5.5.0rc1 since it came out for the quorum queue support. In rc2, 3, 4, and 5, we've been seeing fatal failures on the delayed delivery setup since we have non-durable broadcast queues on the same broker as our durable queues.
> 
> ```
> Traceback (most recent call last):
>   File \"/usr/python39/lib/python3.9/site-packages/celery/worker/worker.py\", line 203, in start
>     self.blueprint.start(self)
>   File \"/usr/python39/lib/python3.9/site-packages/celery/bootsteps.py\", line 116, in start
>     step.start(parent)
>   File \"/usr/python39/lib/python3.9/site-packages/celery/bootsteps.py\", line 365, in start
>     return self.obj.start()
>   File \"/usr/python39/lib/python3.9/site-packages/celery/worker/consumer/consumer.py\", line 341, in start
>     blueprint.start(self)
>   File \"/usr/python39/lib/python3.9/site-packages/celery/bootsteps.py\", line 116, in start
>     step.start(parent)
>   File \"/usr/python39/lib/python3.9/site-packages/celery/worker/consumer/delayed_delivery.py\", line 34, in start
>     bind_queue_to_native_delayed_delivery_exchange(connection, queue)
>   File \"/usr/python39/lib/python3.9/site-packages/kombu/transport/native_delayed_delivery.py\", line 94, in bind_queue_to_native_delayed_delivery_exchange
>     queue.bind_to(exchange.name, routing_key=routing_key)
>   File \"/usr/python39/lib/python3.9/site-packages/kombu/entity.py\", line 681, in bind_to
>     return (channel or self.channel).queue_bind(
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/channel.py\", line 939, in queue_bind
>     return self.send_method(
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/abstract_channel.py\", line 79, in send_method
>     return self.wait(wait, returns_tuple=returns_tuple)
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/abstract_channel.py\", line 99, in wait
>     self.connection.drain_events(timeout=timeout)
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/connection.py\", line 526, in drain_events
>     while not self.blocking_read(timeout):
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/connection.py\", line 532, in blocking_read
>     return self.on_inbound_frame(frame)
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/method_framing.py\", line 53, in on_frame
>     callback(channel, method_sig, buf, None)
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/connection.py\", line 538, in on_inbound_method
>     return self.channels[channel_id].dispatch_method(
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/abstract_channel.py\", line 156, in dispatch_method
>     listener(*args)
>   File \"/usr/python39/lib/python3.9/site-packages/amqp/channel.py\", line 293, in _on_close
>     raise error_for_code(
> amqp.exceptions.NotFound: Queue.bind: (404) NOT_FOUND - no queue 'bcast.3368d7e7-1dcc-4702-b568-02fce1d51952' in vhost '/')
> ```
> 
> I see that you refactored `delayed_delivery.py` a couple of weeks ago on main. I pulled your change into 5.5.0rc5. The failure is less fatal, but still seems wrong.
> 
> ```
> Failed to bind queue 'bcast.f7285996-3ddb-4781-b22c-c86ecb53089a': Queue.bind: (404) NOT_FOUND - no queue 'bcast.f7285996-3ddb-4781-b22c-c86ecb53089a' in vhost '/'
> Failed to bind queues for 'amqp://ph:FPeSxaW1pMd@localhost:5671//': Queue.bind: (404) NOT_FOUND - no queue 'bcast.f7285996-3ddb-4781-b22c-c86ecb53089a' in vhost '/'
> Failed to setup delayed delivery for 'amqp://ph:FPeSxaW1pMd@localhost:5671//': Queue.bind: (404) NOT_FOUND - no queue 'bcast.f7285996-3ddb-4781-b22c-c86ecb53089a' in vhost '/'
> Failed to setup delayed delivery for all broker URLs. Native delayed delivery will not be available.
> db_heavy@7a8cc0a4-2ec8-445d-b816-70fe4ed8be77 ready.
> Started consuming from celery_db_heavy.leader
> ```
> 
> Do you think this is correct and as designed? Seems to us like you should maybe be skipping any `Broadcast` queues in `DelayedDeliever._bind_queues`?

Please open a bug report and tag me, thank you 🙏

**Comment by Nusnus** [comment]
> > > > > > > > > [Celery v5.5.0b1](https://github.com/celery/celery/releases/tag/v5.5.0b1) released.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > [Celery v5.5.0b2](https://github.com/celery/celery/releases/tag/v5.5.0b2) released.
> > > > > > > 
> > > > > > > 
> > > > > > > [Celery v5.5.0b3](https://github.com/celery/celery/releases/tag/v5.5.0b3) released.
> > > > > > 
> > > > > > 
> > > > > > [Celery v5.5.0b4](https://github.com/celery/celery/releases/tag/v5.5.0b4) released.
> > > > > 
> > > > > 
> > > > > [Celery v5.5.0rc1](https://github.com/celery/celery/releases/tag/v5.5.0rc1) released.
> > > > 
> > > > 
> > > > [Celery v5.5.0rc2](https://github.com/celery/celery/releases/tag/v5.5.0rc2) released.
> > > 
> > > 
> > > [Celery v5.5.0rc3](https://github.com/celery/celery/releases/tag/v5.5.0rc3) released.
> > 
> > 
> > [Celery v5.5.0rc4](https://github.com/celery/celery/releases/tag/v5.5.0rc4) released.
> 
> [Celery v5.5.0rc5](https://github.com/celery/celery/releases/tag/v5.5.0rc5) released.

[Celery v5.5.0](https://github.com/celery/celery/releases/tag/v5.5.0) _is finally_ released.

- Announcement on X: https://x.com/CeleryOrg/status/1906813307417510061
- Announcement on LinkedIn: https://www.linkedin.com/posts/celeryorg_release-v550-celerycelery-activity-7312579001259048960-VqJx

LFG! 🚀

**Comment by Nusnus** [comment]
Mission Completed.

P.S
> I'll have to stretch my statement to sun/mon (still March 😉), but yes. There are no new delays since my last update.

In my tz it was just a bit before 00:00 😇
